engine/resources/cron.go

```diff
@@ -462,17 +463,33 @@ func (obj *CronUID) IFF(uid engine.ResUID) bool {
 	if !ok {
 		return false
 	}
-	return obj.name == res.name
+	if obj.unit != res.unit {
+		return false
+	}
+	if obj.session != res.session {
+		return false
+	}
+	return true
+}
+
+// AutoEdges returns the AutoEdge interface.
+func (obj *CronRes) AutoEdges() (engine.AutoEdge, error) {
+	return nil, nil
 }
 
 // UIDs includes all params to make a unique identification of this object.
 // Most resources only return one although some resources can return multiple.
 func (obj *CronRes) UIDs() []engine.ResUID {
+	unit := fmt.Sprintf("%s.service", obj.Name())
+	if obj.Unit != "" {
+		unit = obj.Unit
+	}
 	x := &CronUID{
 		BaseUID: engine.BaseUID{Name: obj.Name(), Kind: obj.Kind()},
-		name:    obj.Name(), // svc name
+		unit:    unit,        // name of target unit
+		session: obj.Session, // user session
 	}
-	return append([]engine.ResUID{x}, obj.file.UIDs()...)
```

Comment:
```panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1412f3a]

goroutine 246 [running]:
github.com/purpleidea/mgmt/engine/traits.(*Named).Name(...)
	/home/jon/go/src/github.com/purpleidea/mgmt/engine/traits/named.go:33
github.com/purpleidea/mgmt/engine/resources.(*FileRes).UIDs(0x0, 0xc0004c3480, 0xc00002b8e0, 0x1)
	/home/jon/go/src/github.com/purpleidea/mgmt/engine/resources/file.go:909 +0x2a
github.com/purpleidea/mgmt/engine/resources.(*CronRes).UIDs(0xc000164b40, 0xc000268f2d, 0xc000164b40, 0x7ff7c8e83388)
	/home/jon/go/src/github.com/purpleidea/mgmt/engine/resources/cron.go:500 +0x1e2
github.com/purpleidea/mgmt/engine/graph/autoedge.addEdgesByMatchingUIDS(0x7ff7c8e83420, 0xc00003d080, 0xc00061cdd0, 0x1, 0x1, 0xc0004c51c0, 0x0, 0xc00002bd50, 0x18, 0x17cbb40, ...)
	/home/jon/go/src/github.com/purpleidea/mgmt/engine/graph/autoedge/autoedge.go:125 +0x32c
github.com/purpleidea/mgmt/engine/graph/autoedge.AutoEdge(0xc0004c51c0, 0xc000574d00, 0xc00002bd50, 0xfc6f64, 0xc0004c51c0)
	/home/jon/go/src/github.com/purpleidea/mgmt/engine/graph/autoedge/autoedge.go:86 +0x45f
github.com/purpleidea/mgmt/engine/graph.(*Engine).AutoEdge(0xc000436140, 0xc00002beb8, 0x0)
	/home/jon/go/src/github.com/purpleidea/mgmt/engine/graph/autoedge.go:29 +0x5a
github.com/purpleidea/mgmt/lib.(*Main).Run.func10(0x19c92b8, 0xc000392b24, 0xc0001c4480, 0xc0004d4c80, 0xc0001c4488, 0xc0000bd200, 0xc0001ea380, 0x6, 0xc000592480, 0x1bece00, ...)
	/home/jon/go/src/github.com/purpleidea/mgmt/lib/main.go:593 +0x7c5
created by github.com/purpleidea/mgmt/lib.(*Main).Run
	/home/jon/go/src/github.com/purpleidea/mgmt/lib/main.go:423 +0xd7c

shell returned 2
```

Reply:
also, obj.file will be empty in certain cases

Reply:
Do we want to include it when it's not nil?

Reply:
This panic was not when it was nil. I think it's because obj.file isn't a fully implemented resource unto itself. Maybe you can tell me what I'm missing. ;)